public/calculator.js

```diff
@@ -0,0 +1,145 @@
+class Calculator {
+	constructor(previousOperandTextElement, currentOperandTextElement, lastAnswerElement) {
+		this.previousOperandTextElement = previousOperandTextElement;
+		this.currentOperandTextElement = currentOperandTextElement;
+		this.lastAnswerElement = lastAnswerElement;
+		this.clear;
+	}
+	clear() {
+		this.currentOperand = "";
+		this.previousOperand = "";
+		this.operation = undefined;
+	}
+	delete() {
+		this.currentOperand = this.currentOperand.toString().slice(0 - 1);
+	}
+	appendNumber(number) {
+		if (number === "." && this.currentOperand.includes(".")) return;
+		this.currentOperand = this.currentOperand?.toString() + number.toString();
+	}
+	chooseOperation(operation) {
+		if (this.currentOperand === "") return;
+		if (this.previousOperand !== "") {
+			this.compute();
+		}
+		this.operation = operation;
+		this.previousOperand = this.currentOperand;
+		this.currentOperand = "";
+	}
+	compute() {
+		let computation;
+		const prev = parseFloat(this.previousOperand);
+		const current = parseFloat(this.currentOperand);
+		if (isNaN(prev) || isNaN(current)) return;
+		switch (this.operation) {
+			case "+":
+				computation = prev + current;
+				break;
+			case "-":
+				computation = prev - current;
+				break;
+			case "*":
+				computation = prev * current;
+				break;
+			case "÷":
+				computation = prev / current;
+				break;
+			default:
+				return;
+		}
+		this.currentOperand = computation;
+		this.operation = undefined;
+		this.previousOperand = "";
+		axios.post("/save-last-answer", {
+			lastAnswer: this.currentOperand
+		});
+		lastAnswerElement.innerText = this.currentOperand;
```

Comment:
When an answer is computed we POST the answer to the server to save it to the `last-answer.dat` file and we update our last answer `<span>` element to display it.